src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,236 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.util.DateTimeHandler;
+
+/**
+ * Encapsulates the handling of user's input.
+ */
+public class Parser {
+
+    /** Storage for parser to write data to. */
+    private final Storage storage;
+
+    /** UI for parser to display messages. */
+    private final Ui ui;
+
+    /** Task list for parser to add/edit/delete tasks. */
+    private final TaskList taskList;
+
+    /**
+     * Creates a new parser from a storage.
+     *
+     * @param storage Storage of Duke's data.
+     */
+    public Parser(Storage storage) {
+        this.storage = storage;
+        this.ui = this.storage.getUi();
+        this.taskList = this.storage.getTaskList();
+    }
+
+    /**
+     * Processes the input fed to Duke.
+     *
+     * @param input Input string.
+     * @throws DukeException If input is invalid.
+     */
+    public void processInput(String input) throws DukeException {
+        if (input.equals("bye")) {
+            this.ui.printMessage("Bye. Hope to see you again soon!");
+        } else if (input.equals("list")) {
+            this.ui.printTaskList(this.taskList);
+        } else if (input.contains("delete")) {
```

Comment:
Not sure if I'm supposed to give this type of comments, but I think if your user input is something like `todo delete files` your parser will run the wrong command :(

Reply:
Will fix possibility of user running two or more of such commands stated in this method.